sdk/hooks.go

```diff
@@ -0,0 +1,211 @@
+package sdk
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/okta/okta-sdk-golang/okta"
+	"github.com/okta/okta-sdk-golang/okta/query"
+)
+
+type (
+	AuthScheme struct {
+		Key   string `json:"key,omitempty"`
+		Type  string `json:"type,omitempty"`
+		Value string `json:"value,omitempty"`
+	}
+
+	Channel struct {
+		Config  *HookConfig `json:"config"`
+		Type    string      `json:"type,omitempty"`
+		Version string      `json:"version,omitempty"`
+	}
+
+	HookConfig struct {
+		AuthScheme *AuthScheme `json:"authScheme,omitempty"`
+		Headers    []*Header   `json:"headers,omitempty"`
+		URI        string      `json:"uri,omitempty"`
+		Method     string      `json:"method,omitempty"`
+	}
+
+	Header struct {
+		Key   string `json:"key,omitempty"`
+		Value string `json:"value,omitempty"`
+	}
+
+	Events struct {
+		Items []string `json:"items"`
+	}
+
+	EventHook struct {
+		Channel *Channel `json:"channel"`
+		ID      string   `json:"id,omitempty"`
+		Name    string   `json:"name,omitempty"`
+		Status  string   `json:"status,omitempty"`
+		Events  *Events  `json:"events,omitempty"`
+	}
+
+	InlineHook struct {
+		Channel *Channel `json:"channel"`
+		ID      string   `json:"id,omitempty"`
+		Name    string   `json:"name,omitempty"`
+		Status  string   `json:"status,omitempty"`
+		Type    string   `json:"type,omitempty"`
+		Version string   `json:"version,omitempty"`
+	}
+)
+
+func (e *Events) MarshalJSON() ([]byte, error) {
```

Comment:
This makes the `Events` type to implement the [`Marshaler`](https://golang.org/pkg/encoding/json/#Marshaler) interface so that `json.Marshal` calls this function when marshaling an `Events` object.  It's here just to inject static value `"EVENT_TYPE"` as the `type` key in the marshaled JSON object since that key should never have any other value (currently, at least).

Looking at it now, it may be overly-clever.  I can add an explicit `Type` member to the `Events` class and get rid of this.